src/ArchiveAdmin.php

```diff
@@ -294,12 +294,16 @@ public function getManagedModelTabs()
 
         $mainModels = $this->getVersionedModels('main', true);
         foreach ($mainModels as $class => $title) {
-            $forms->push(new ArrayData(array(
-                'Title' => $title,
-                'ClassName' => $class,
-                'Link' => $this->Link($this->sanitiseClassName($class)),
-                'LinkOrCurrent' => ($class === $this->modelClass) ? 'current' : 'link'
-            )));
+            if (ClassInfo::hasMethod(Injector::inst()->get($class), 'isArchiveFieldEnabled')
+                && Injector::inst()->get($class)->isArchiveFieldEnabled()
```

Comment:
Side note: the duplicated use of Injector::inst()->get() could be avoided

Reply:
It's now avoided